tools/pkg/engine_build_configs/lib/src/build_config_runner.dart

```diff
@@ -289,67 +288,157 @@ final class GlobalBuildRunner extends Runner {
     return result.ok;
   }
 
-  // TODO(zanderso): This should start and stop RBE when it is an --rbe build.
-  Future<bool> _runNinja(RunnerEventHandler eventHandler) async {
-    final String ninjaPath = p.join(
-      engineSrcDir.path, 'flutter', 'third_party', 'ninja', 'ninja',
+  late final String _hostCpu = (){
+    if (platform.isWindows) {
+      return platform.environment['PROCESSOR_ARCHITECTURE'] ?? 'x64';
+    }
+    final List<String> unameCommand = <String>['uname', '-m'];
+    final io.ProcessResult unameResult = processRunner.processManager.runSync(
+      unameCommand,
     );
-    final String outDir = p.join(engineSrcDir.path, 'out', build.ninja.config);
-    final List<String> command = <String>[
-      ninjaPath,
-      '-C', outDir,
-      if (_isGomaOrRbe) ...<String>['-j', '200'],
-      ...extraNinjaArgs,
-      ...build.ninja.targets,
-    ];
-    eventHandler(RunnerStart('${build.name}: ninja', command, DateTime.now()));
+    return unameResult.exitCode == 0 ? (unameResult.stdout as String).trim() : 'x64';
+  }();
```

Comment:
I'd consider using `Abi.current` for this: https://api.dart.dev/stable/3.2.6/dart-ffi/Abi-class.html